x/config/spec.go

```diff
@@ -217,6 +223,27 @@ type TransportSpec struct {
 	// the first thing inside their BuildInbound.
 }
 
+// PeerListPreset defines a named preset for a peer list. Peer list presets
+// may be used by specifying a `with` key in the outbound configuration.
+//
+// 	http:
+// 	  with: mypreset
+type PeerListPreset struct {
+	Name string
+
+	// A function in the shape,
+	//
+	//  func(peer.Transport, *config.Kit) (peer.Chooser, error)
```

Comment:
wait, I'm a bit confused, should this be a "peer.Transport" or should it be a "http.Transport"?

If it's the peer.Transport, is there any reason we shouldn't make the BuildPeerList have the exact function signature?